src/main/java/com/atlassian/db/replica/api/reason/RouteDecision.java

```diff
@@ -38,14 +42,31 @@ public Reason getReason() {
         return Optional.ofNullable(cause);
     }
 
+    /**
+     * @return information whether reason was caused by write sql operation.
```

Comment:
A JavaDoc ❤️

Reply:
I would define a "write" like this:
```suggestion
     * @return true if the database call would always fail on a read replica
```

Reply:
> true if the database call would always fail on a read replica
@dagguh interesting suggestion to define what is Write 

Are we sure about always? or always on specific db-replica version?

Reply:
I'm afraid we don't have implemented logic that will determine this.

Reply:
Always = not-transient, retrying will not help.